azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -151,6 +151,37 @@ def create_from_x509_certificate(cls, x509, hostname, device_id, **kwargs):
         iothub_pipeline = pipeline.IoTHubPipeline(authentication_provider, pipeline_configuration)
         return cls(iothub_pipeline)
 
+    @classmethod
+    def create_from_registration_result_and_symmetric_key(
```

Comment:
>create_from_registration_result_and_symmetric_key [](start = 8, length = 49)

I don't have a better answer, but this API (and the x509 one) both feel clunky.  This is much easier than making the user build a connection string, but it feels slapped together.

Reply:
I agree, I was under the impression that a registration result was sufficient. 

---
In reply to: [354670450](https://github.com/Azure/azure-iot-sdk-python/pull/407#discussion_r354670450) [](ancestors = 354670450)

Reply:
yes it feels like that. The other thing that can be done is to somehow give back the key as part of registration result but that would also mean that x509 becomes an attribute of the result as well. But it does not really make sense as the user already has those

---
In reply to: [354962358](https://github.com/Azure/azure-iot-sdk-python/pull/407#discussion_r354962358) [](ancestors = 354962358,354670450)